src/scenes/AcceleratorRouter/AcceleratorDeliverablesView.tsx

```diff
@@ -1,23 +1,112 @@
+import { useCallback, useMemo, useState } from 'react';
 import { useParams } from 'react-router-dom';
+import { Box } from '@mui/material';
+import { Button, DropdownItem } from '@terraware/web-components';
+import strings from 'src/strings';
+import { DeliverableStatusType } from 'src/types/Deliverables';
 import Page from 'src/components/Page';
+import OptionsMenu from 'src/components/common/OptionsMenu';
 import DeliverableView from 'src/components/DeliverableView';
 import useFetchDeliverable from 'src/components/DeliverableView/useFetchDeliverable';
+import useEditStatusDeliverable from './useEditStatusDeliverable';
+import RejectDialog from './RejectDialog';
 
 const AcceleratorDeliverablesView = () => {
+  const [showRejectDialog, setShowRejectDialog] = useState<boolean>(false);
   const { deliverableId } = useParams<{ deliverableId: string }>();
+  const { status: requestStatus, update } = useEditStatusDeliverable();
 
   const { deliverable } = useFetchDeliverable({
     deliverableId: Number(deliverableId),
     isAcceleratorConsole: true,
   });
 
+  // temporary solution until we have the confirmation modal design
+  const setStatus = useCallback(
+    (status: DeliverableStatusType) => {
+      if (deliverable?.id !== undefined) {
+        if (window.confirm(`Are you sure you want to set the status to ${status} ?`)) {
+          update({ id: deliverable.id, status });
+        }
+      }
+    },
+    [deliverable?.id, update]
+  );
+
+  const rejectDeliverable = useCallback(
+    (reason: string) => {
+      if (deliverable?.id !== undefined) {
+        update({ id: deliverable.id, status: 'Rejected', reason });
+      }
+      setShowRejectDialog(false);
+    },
+    [deliverable?.id, setShowRejectDialog, update]
+  );
+
+  const onOptionItemClick = useCallback(
+    (optionItem: DropdownItem) => {
+      switch (optionItem.value) {
+        case 'not_submitted': {
+          setStatus('Not Submitted');
+          break;
+        }
+        case 'needs_translation': {
+          setStatus('Needs Translation');
+          break;
+        }
+        case 'not_needed': {
+          setStatus('Not Needed');
+          break;
```

Comment:
These values are passed into the update request, shouldn't they be the `optionItem.value` instead of the "readable" version? Maybe you've got it this way because there aren't backend types, disregard if that's the case